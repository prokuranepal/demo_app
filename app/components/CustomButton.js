import React from 'react';
import { View, TouchableOpacity, Text, StyleSheet } from 'react-native';

import PropTypes from 'prop-types';

import {fontSize} from '../theme/fonts';
import {colors} from '../theme/themeColors';

const CustomButton = (props) => (
  <View style={props.container_style}>
    <TouchableOpacity
      style={[styles.saveStyle, props.buttonStyle]}
      onPress={props.pressHandler}
      data-test="touchableComp">
      <Text style={{ ...styles.text,color: props.color}} data-test="textData">{props.title}</Text>
    </TouchableOpacity>
  </View>
)

const styles = StyleSheet.create({
  text: {
    fontSize: fontSize.buttonText
  },
  saveStyle: {
    borderRadius: 40,
    paddingHorizontal: 60,
    paddingVertical: 10,
    backgroundColor: colors.primaryButton
  }
})

CustomButton.propTypes = {
  buttonStyle: PropTypes.object,
  containerStyle: PropTypes.object,
  color: PropTypes.string,
  title: PropTypes.string,
  pressHandler: PropTypes.func
}


export default CustomButton;