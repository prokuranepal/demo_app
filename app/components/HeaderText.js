import React from "react";
import { Text, StyleSheet } from "react-native";

import PropTypes from 'prop-types';

import {fontSize} from '../theme/fonts';
import {colors} from '../theme/themeColors'

const HeaderText = props => (
  <Text {...props} style={[styles.textHeading, props.style]} data-test="headerText">
    {props.children}
  </Text>
);

const styles = StyleSheet.create({
  textHeading: {
    fontSize: fontSize.header,
    color: colors.textColor1,
    fontWeight: "bold"
  }
});

HeaderText.propTypes = {
  style: PropTypes.object,
  children: PropTypes.any
}

export default HeaderText;