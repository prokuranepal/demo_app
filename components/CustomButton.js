import React from 'react';
import { View, Button, TouchableOpacity, Text,StyleSheet } from 'react-native';
const customButton =(props)=>(
    <View style={props.container_style}>
        <TouchableOpacity
          style={[styles.saveStyle,props.buttonStyle]}
          onPress={ props.pressHandler }>
            <Text style={{color:props.color, fontSize:17}}> {props.title} </Text>
      </TouchableOpacity>
</View>
)

const styles = StyleSheet.create({
saveStyle:{
    borderRadius:40,
    paddingHorizontal:60,
    paddingVertical:10,
    backgroundColor:"#4fbed2"
  }
})

export default customButton;