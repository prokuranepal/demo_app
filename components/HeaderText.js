import React from "react";
import { Text, StyleSheet } from "react-native";
import {colors} from '../ThemeColors/themeColors'
const headerText = props => (
  <Text {...props} style={[styles.textHeading, props.style]}>
    {props.children}
  </Text>
);

const styles = StyleSheet.create({
  textHeading: {
    fontSize: 26,
    color: colors.textColor1,
    fontWeight: "bold"
  }
});

export default headerText;