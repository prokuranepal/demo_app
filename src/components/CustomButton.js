import React from 'react';
import { View, Button, TouchableOpacity, Text, StyleSheet } from 'react-native';
import {colors} from '../ThemeColors/themeColors';
const customButton = (props) => (
  <View style={props.container_style}>
    <TouchableOpacity
      style={[styles.saveStyle, props.buttonStyle]}
      onPress={props.pressHandler} data-test="touchableComp">
      <Text style={{ color: props.color, fontSize: 17 }} data-test="textData">{props.title}</Text>
    </TouchableOpacity>
  </View>
)

const styles = StyleSheet.create({
  saveStyle: {
    borderRadius: 40,
    paddingHorizontal: 60,
    paddingVertical: 10,
    backgroundColor: colors.primaryButton
  }
})

export default customButton;