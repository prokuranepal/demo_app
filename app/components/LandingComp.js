

import React from 'react';
import {
    StyleSheet,
    View,
} from 'react-native';

import PropTypes from 'prop-types';

import CustomButton from './CustomButton';
import HeaderText from './HeaderText';

import {fontSize} from '../theme/fonts';
import {colors} from '../theme/themeColors';


const LandingComp = (props) => (
    <View style={{ ...styles.sectionContainer, backgroundColor: props.backgroundCol }}>
        <HeaderText style={{color: colors.textColor2}} data-test="headComp">
            Let's know your dietary preferences.
        </HeaderText>
        <HeaderText style={styles.description} data-test="descComp">
            Any ingredients you don't prefer or are allergic to?
        </HeaderText>
        {props.children}
        <CustomButton container_style={styles.buttonContainer}
            data-test="buttonComp"
            title="save diet"
            color={colors.buttonTextColor}
            pressHandler={() => props.setisModal(true)} />
    </View>
)


const styles = StyleSheet.create({

    description:{fontSize: fontSize.buttonText, paddingTop: 10},
    buttonContainer: {
        flex: 1,
        justifyContent: 'flex-end',
        alignItems: 'center',
        marginBottom: 40,
    },

    sectionContainer: {
        flex: 1,
        marginTop: 64,
        paddingHorizontal: 48,
        backgroundColor: colors.section
    },

});

LandingComp.propTypes = {
    backgroundCol : PropTypes.string,
    children: PropTypes.any,
    setisModal: PropTypes.func
  }

export default LandingComp;