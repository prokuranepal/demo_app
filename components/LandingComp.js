

import React, {useState} from 'react';
import {
    StyleSheet,
    View,
} from 'react-native';
import CustomButton from './CustomButton';
import HeaderText from './HeaderText';


const LandingComp =(props)=>(
    <View style={{...styles.sectionContainer,backgroundColor:props.backgroundCol}}>
        <HeaderText style={{color:"#5a638b"}}>
            Let's know your dietary preferences.
        </HeaderText>
        <HeaderText style={{fontSize:17, paddingTop:10}}>
            Any ingredients you don't prefer or are allergic to?
        </HeaderText>
        {props.children}
        <CustomButton container_style ={styles.buttonContainer} 
                data-test="callComp" 
                title="save diet"  
                color="#fff"
                pressHandler={() => props.setisModal(true)} />
    </View>
)


const styles = StyleSheet.create({

   buttonContainer: {
        flex:1,
        justifyContent: 'flex-end',
        alignItems: 'center',
        marginBottom: 40,
    },

    sectionContainer: {
        flex:1,
        marginTop: 64,
        paddingHorizontal: 48,
        backgroundColor:"#f3f3fb"
    },

});

export default LandingComp;