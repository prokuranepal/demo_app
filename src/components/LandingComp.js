

import React, { useState } from 'react';
import {
    StyleSheet,
    View,
} from 'react-native';
import CustomButton from './CustomButton';
import HeaderText from './HeaderText';
import {colors} from '../ThemeColors/themeColors';

const LandingComp = (props) => (
    <View style={{ ...styles.sectionContainer, backgroundColor: props.backgroundCol }}>
        <HeaderText style={{color: colors.textColor2}} data-test="headComp">
            Let's know your dietary preferences.
        </HeaderText>
        <HeaderText style={{ fontSize: 17, paddingTop: 10 }} data-test="descComp">
            Any ingredients you don't prefer or are allergic to?
        </HeaderText>
        {props.children}
        <CustomButton container_style={styles.buttonContainer}
            data-test="buttonComp"
            title="save diet"
            color={colors.buttonTextColor}
            pressHandler={() => props.setisModal(true)} />
    </View>
)


const styles = StyleSheet.create({

    buttonContainer: {
        flex: 1,
        justifyContent: 'flex-end',
        alignItems: 'center',
        marginBottom: 40,
    },

    sectionContainer: {
        flex: 1,
        marginTop: 64,
        paddingHorizontal: 48,
        backgroundColor: "#f3f3fb"
    },

});

export default LandingComp;