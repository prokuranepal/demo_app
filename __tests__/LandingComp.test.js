import React from 'react';
import {
    configure,
    shallow
} from 'enzyme';
import {View,Text} from 'react-native';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import HeaderText from '../src/components/HeaderText';
import CustomButton from '../src/components/CustomButton';
import LandingComp from '../src/components/LandingComp';


import {findByTestAttr} from '../src/utils/testFunction';
// JestHook.mock('expo-font');
configure({
    adapter: new EnzymeAdapter
})

const function_click=jest.fn()
const dummy_data = {
    setisModal:function_click
}



describe('<LandingComp/>', () => {

    it('should render without issues', () => {
        const component = shallow(<LandingComp />);
        expect(component.length).toBe(1);
        expect(component).toMatchSnapshot();
      });

    it("Individual Components render test in LandingComp", () =>
      {
          const wrapper = shallow(<LandingComp>Headerrr</LandingComp>)
          expect(wrapper.find(View)).toHaveLength(1)
          expect(wrapper.find(HeaderText)).toHaveLength(2)
          expect(wrapper.find(CustomButton)).toHaveLength(1)
      
     });

     it("Props and event test in LandingComp", () =>
     {
         const wrapper = shallow(<LandingComp {...dummy_data}><Text data-test="childComp"/></LandingComp>)
         let childComp = findByTestAttr(wrapper, "childComp")
         expect(childComp.length).toBe(1);
         let headComp = findByTestAttr(wrapper, "headComp")
         expect(headComp.length).toBe(1);         
         expect(headComp.props().children).toEqual("Let's know your dietary preferences.")
        
         let descComp = findByTestAttr(wrapper, "descComp")
         expect(descComp.length).toBe(1);         
         expect(descComp.props().children).toEqual("Any ingredients you don't prefer or are allergic to?")
        
         let buttonComp = findByTestAttr(wrapper, "buttonComp")
         expect(buttonComp.props().title).toEqual("save diet")
         buttonComp.props().pressHandler()
         wrapper.update()
         expect(function_click).toHaveBeenCalledTimes(1)
         expect(function_click).toHaveBeenCalledWith(true)
     
    });


})