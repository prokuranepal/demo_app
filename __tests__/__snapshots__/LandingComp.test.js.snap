// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LandingComp/> should render without issues 1`] = `ShallowWrapper {}`;
